src/lib.rs

```diff
@@ -545,7 +570,21 @@ impl Iter {
         let rd = fs::read_dir(dent.path()).map_err(|err| {
             Some(Error::from_path(self.depth, dent.path().to_path_buf(), err))
         });
-        self.stack_list.push(DirList::Opened { depth: self.depth, it: rd });
+        let mut list = DirList::Opened { depth: self.depth, it: rd };
+        if let Some(ref mut cmp) = self.opts.sorter {
+            let mut entries = list.collect::<Vec<_>>();
+            entries.sort_by(|ref a, ref b| {
```

Comment:
I think this closure can take `a` and `b` by value, right? That would get rid of the nested `&&` in the `match` below I think.